src/domain/common/whiteboard/excalidraw/useWhiteboardFilesManager.ts

```diff
@@ -0,0 +1,257 @@
+import { useMemo, useRef, useState } from 'react';
+import { useUploadFileMutation } from '../../../../core/apollo/generated/apollo-hooks';
+import { BinaryFileData, DataURL, ExcalidrawAPIRefValue } from '@alkemio/excalidraw/types/types';
+import { excalidrawFileMimeType, generateIdFromFile } from './collab/utils';
+
+const isValidDataURL = (url: string) =>
+  url.match(/^(data:)([\w/+-]*)(;charset=[\w-]+|;base64){0,1},[A-Za-z0-9+/=]+$/gi) !== null;
+
+const dataUrlToFile = async (
+  dataUrl: string,
+  fileName: string = 'from data',
+  mimeType: string | undefined = undefined,
+  lastModified: number = new Date().getTime()
+): Promise<File> => {
+  if (!isValidDataURL(dataUrl)) {
+    return Promise.reject('Not a valid dataURL detected');
+  }
+  const mime = dataUrl.split(',')?.[0]?.match(/:(.*?);/)?.[1];
+  const blob = await (await fetch(dataUrl)).blob();
+  var file = new File([blob], fileName, { type: mime ?? mimeType ?? 'application/octet-stream', lastModified });
+  return file;
+};
+
+const blobToDataURL = (blob: Blob): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      resolve(reader.result as string);
+    };
+    reader.onerror = () => {
+      reject(reader.error);
+    };
+    reader.readAsDataURL(blob);
+  });
+};
+
+const fetchFileToDataURL = async (url: string): Promise<string> => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch file from ${url}`);
+  }
+  const blob = await response.blob();
+  return blobToDataURL(blob);
+};
+
+export type BinaryFileDataWithUrl = BinaryFileData & { url?: string };
+
+interface Props {
+  storageBucketId?: string; // FilesManagers without storageBucketId will throw an exception on file upload
+  excalidrawApi: ExcalidrawAPIRefValue | null;
+  allowFallbackToAttached?: boolean;
+}
+
+interface WhiteboardWithFiles {
+  files?: Record<string, BinaryFileDataWithUrl>;
+}
+export interface WhiteboardFilesManager {
+  addNewFile: (file: File) => Promise<string>;
+  loadFiles: (data: WhiteboardWithFiles) => Promise<void>;
+  pushFilesToExcalidraw: () => Promise<void>;
+  removeExcalidrawAttachments: <W extends WhiteboardWithFiles>(whiteboard: W) => Promise<W>;
+  loading: {
+    uploadingFile: boolean;
+    downloadingFiles: boolean;
+  };
+}
+
+const useWhiteboardFilesManager = ({
+  storageBucketId,
+  allowFallbackToAttached,
+  excalidrawApi,
+}: Props): WhiteboardFilesManager => {
+  const log = (..._args) => {
+    // TODO: Remove those `log()`s when this is confirmed to be fully stable
```

Comment:
todo

Reply:
I would like these ones to stay. I have a few `log`s in this file that will not log anything when deployed to the server because this function does nothing but for debugging locally I can easily uncomment this line. 
I have added another requirement to #4821 to remove them in this sprint